interp/interp.go

```diff
@@ -0,0 +1,176 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package interp
+
+import (
+	"fmt"
+	"math"
+	"sort"
+)
+
+// Interpolator1D represents a 1D interpolator.
+// It interpolates a y(x) function based on some data over a range [begin(), end()].
+// Both begin() and end() can be infinite, but begin() must be lower than end().
+type Interpolator1D interface {
+	// Returns the lowest allowed argument for evaluate().
+	begin() float64
```

Comment:
perhaps lump `begin` and `end` together into, say, `Interval`, that would return a `r1.Interval` from http://godoc.org/gonum.org/v1/gonum/spatial/r1#Interval ?

Reply:
Will do.